e2e/ls_test.go

```diff
@@ -391,3 +393,204 @@ func TestListS3ObjectsWithDashH(t *testing.T) {
 		1: match(`^ 264.0K testfile2.txt$`),
 	}, trimMatch(dateRe), alignment(true))
 }
+
+// ls --exclude "main*" s3://bucket/
+func TestListS3ObjectsWithExcludeFilter(t *testing.T) {
```

Comment:
Could you also add a test similar to this but lists bucket with a wildcard pattern to assert it also excludes matching files under prefixes -pseudo directories-, like `a/*.txt` files for this test.

Reply:
added test case in other [PR](https://github.com/peak/s5cmd/pull/334).